### PR TITLE
Adding section on type view.

### PR DESCRIPTION
**Pull request by gusthoff**

### CODE REVIEW

File: content/courses/advanced-ada/chapters/types.rst [hunk 1]
```diff
@@ -445,10 +445,123 @@ Mutually dependent types
...
+    package Full_View is
+
+       --  Full view of the R type:
+       type R is record
+          I : Integer;
+       end record;
+
+    end Full_View;
+
+As soon as we start applying encapsulation |mdash| via the :ada:`private`
```
Comment (pat-rogers) [inline content/courses/advanced-ada/chapters/types.rst:464]: encapsulation and information hiding

File: content/courses/advanced-ada/chapters/types.rst [hunk 2]
```diff
@@ -445,10 +445,123 @@ Mutually dependent types
...
+keyword |mdash| to a specific type, we need to split the type declaration into
+a partial view and a full view. For example:
```
Comment (pat-rogers) [inline content/courses/advanced-ada/chapters/types.rst:466]: Maybe something like:
"a specific type, we are introducing a partial view and making only that view compile-time visible to clients. Doing so requires us to introduce the private part of the package (unless already present)."

Comment (gusthoff) [reply content/courses/advanced-ada/chapters/types.rst]: Yes, I like it.

File: content/courses/advanced-ada/chapters/types.rst [hunk 3]
```diff
@@ -445,10 +445,125 @@ Mutually dependent types
...
+       --  Partial view of R indicates that R exists as a private type,
+       --  so we can declare objects of this type:
+       C : R;
+    begin
+       --  But we cannot directly access any information declared in the full
+       --  view of R:
+       --
+       --  C.I := 42;
+       --
+       null;
```
Comment (pat-rogers) [inline content/courses/advanced-ada/chapters/types.rst:516]: Maybe should have a comment on the line doing the assignment, saying it is illegal

**Review by pat-rogers — APPROVED** [review]

**Pull request merged**